File: exit_hook/exit_hook.c
#include <stdio.h>
#include <stdlib.h>

void cleanup(void) {
    printf("Exiting. Thank you!");
}

int main(int argc, char **argv) {
    atexit(cleanup);    // this function gets called on exit
                        // not called on abort() or crashes
                        // can have multiple atexit()
                        // maximum 32 function pointers.
    
    if (argc != 2) {
        printf("Usage: %s <number>\n", argv[0]);
        return EXIT_FAILURE;
    }
    
    int value = atoi(argv[1]);
    printf("%d^2 == %d\n", value, value*value);
    return EXIT_SUCCESS;
}

File: exit_hook/exit_hook.py
# python code in c repository 
# because python is an abstraction of c code :p

import atexit
import sys

def exit_func():
    print("Exiting. Thank you!")

atexit.register(exit_func)

if (len(sys.argv) != 2):
    print(f"Usage: {sys.argv[0]} <number>")
    sys.exit(1)

value = int(sys.argv[1])
squared = value**2

print(f"{value}^2 == {squared}")